cbits/ctermid.c

```diff
@@ -0,0 +1,21 @@
+/* Provide ctermid() if it is missing (as on some Android versions). */
+
+#include "HsUnixConfig.h"
+#ifndef HAVE_CTERMID
```

Comment:
Fwiw, I consider the use of `#ifdef`s for boolean macros bad style; I know that the `unix` package currently still has a lot of `#ifdef HAVE_*` around, but I'm considering replacing those at some point